src/System.Management.Automation/engine/hostifaces/History.cs

```diff
@@ -1041,114 +1019,137 @@ public string Id
         /// </summary>
         protected override void EndProcessing()
         {
-            // Invoke-history can execute only one command. If multiple
-            // ids were provided, throw exception
-            if (_multipleIdProvided)
-            {
-                ThrowTerminatingError(
-                    new ErrorRecord(
-                        new ArgumentException(HistoryStrings.InvokeHistoryMultipleCommandsError),
-                        "InvokeHistoryMultipleCommandsError",
-                        ErrorCategory.InvalidArgument,
-                        targetObject: null));
-            }
-
             var ctxRunspace = (LocalRunspace)Context.CurrentRunspace;
             History history = ctxRunspace.History;
             Dbg.Assert(history != null, "History should be non null");
 
-            // Get the history entry to invoke
-            HistoryInfo entry = GetHistoryEntryToInvoke(history);
-            string commandToInvoke = entry.CommandLine;
-
-            if (!ShouldProcess(commandToInvoke))
+            // If the user didn't provide an Id, get the ID for the most
+            // recent entry
+            if (Id == null)
             {
-                return;
-            }
+                try
+                {
+                    Id = new string[1] { history.GetEntries(0, 1, true)[0].Id.ToString() };
```

Comment:
nit: named arguments for `history.GetEntries()`

Reply:
Hm, I see your point. Is this a style thing up for debate or not? I get why it's worthwhile to do for non-obvious things like `null` but I wonder if it really adds any notable value in a situation like this?

Reply:
🤷 I tend to prefer it myself, but defer to others here whether it's strictly necessary.

At a glance, though, I can only infer two out of three of those arguments from context.

Reply:
@thomasrayner It is minor comment and It's up to you.